hazelcast/src/main/java/com/hazelcast/config/alto/AltoSocketConfig.java

```diff
@@ -0,0 +1,153 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config.alto;
+
+import com.hazelcast.spi.annotation.Beta;
+
+import javax.annotation.Nonnull;
+import java.util.Objects;
+
+import static com.hazelcast.internal.util.Preconditions.checkNotNull;
+import static com.hazelcast.internal.util.Preconditions.checkPositive;
+
+/**
+ * Socket configuration for Alto. In Alto, each eventloop has its own
+ * sockets.
+ *
+ * @see AltoConfig
+ * @since 5.3
+ */
+@SuppressWarnings("checkstyle:JavadocVariable")
+@Beta
+public class AltoSocketConfig {
+    public static final int DEFAULT_RECEIVE_BUFFER_SIZE_KB = 128;
+    public static final int DEFAULT_SEND_BUFFER_SIZE_KB = 128;
+
+    private String portRange = "11000-21000";
+    private int receiveBufferSizeKB = DEFAULT_RECEIVE_BUFFER_SIZE_KB;
+    private int sendBufferSizeKB = DEFAULT_SEND_BUFFER_SIZE_KB;
+
+    /**
+     * Gets the possible port range for Alto sockets to bind. Can't return
+     * null.
+     *
+     * @return the port range string
+     */
+    @Nonnull
+    public String getPortRange() {
+        return portRange;
+    }
+
+    /**
+     * Sets the possible port range for Alto sockets to bind. Can't return
+     * null.
+     *
+     * @param portRange the port range to set
+     * @return this Alto socket config
+     * @throws IllegalArgumentException if portRange doesn't match {@code
+     *                                  \d{1,5}-\d{1,5}} regular expression
+     * @throws NullPointerException     if portRange is null
+     */
+    @Nonnull
+    public AltoSocketConfig setPortRange(@Nonnull String portRange) {
+        checkNotNull(portRange);
+        if (!portRange.matches("\\d{1,5}-\\d{1,5}")) {
+            throw new IllegalArgumentException("Invalid port range");
+        }
+
+        this.portRange = portRange;
+        return this;
+    }
+
+    /**
+     * Gets the receive-buffer size of the Alto sockets in kilobytes.
+     *
+     * @return the receive-buffer size of the Alto sockets in kilobytes
+     * @see java.net.SocketOptions#SO_RCVBUF
+     */
+    public int getReceiveBufferSizeKB() {
+        return receiveBufferSizeKB;
+    }
+
+    /**
+     * Sets the receive-buffer size of the Alto sockets in kilobytes. Can't
+     * return null.
+     *
+     * @param receiveBufferSizeKB the receive-buffer size of the Alto sockets in kilobytes
+     * @return this Alto socket config
+     * @throws IllegalArgumentException if receiveBufferSizeKB isn't positive
+     * @see java.net.SocketOptions#SO_RCVBUF
+     */
+    @Nonnull
+    public AltoSocketConfig setReceiveBufferSizeKB(int receiveBufferSizeKB) {
+        this.receiveBufferSizeKB = checkPositive("receiveBufferSizeKB", receiveBufferSizeKB);
```

Comment:
The thing is, endpoint config socket size defines classic hz socket buffer sizes. For example user may want to define a smaller socket buffer for generic operations (non partitioned operations for now) and larger socket buffer for alto operations.

Reply:
Perhaps we should just simplify. THe less a user can configure, generally, the better it is.

Reply:
I can also address this under 23775. But lets discuss this thoroughly first. I introduced these just because we found this distinction useful previously. It's also useful to see all alto socket related configurations in one place.